### ISSUE TITLE
Training time far exceeded 30 minutes

### ISSUE THREAD

**Issue by flashcp** [Deep-Agent/R1-V#39]
Hello, thank you for your work. 
While reproducing the results, I found that the training time far exceeded 30 minutes. Could you please explain why this might be the case?
`1%|▉        | 100/17500 [36:26<87:08:16, 18.03s/it]`

I used 4 A100 GPUs, and my training script and Python packages are as follows.

export DEBUG_MODE="false" # Enable Debug if you want to see the rollout of model during RL
export LOG_PATH="./debug_log_2b.txt"
dataset=/nfs/code/R1-V/datasets/leonardPKU___clevr_cogen_a_train

$torchrun --nproc_per_node="4" \
    --nnodes="1" \
    --node_rank="0" \
    --master_addr="127.0.0.1" \
    --master_port="12345" \
    $root/src/open_r1/grpo.py \
    --output_dir $output \
    --model_name_or_path Qwen/Qwen2-VL-2B-Instruct \
    --dataset_name $dataset \
    --max_prompt_length 1024 \
    --per_device_train_batch_size 1 \
    --gradient_accumulation_steps 2 \
    --logging_steps 1 \
    --bf16 \
    --report_to tensorboard \
    --gradient_checkpointing false \
    --attn_implementation flash_attention_2 \
    --max_pixels 401408 \
    --num_train_epochs 2 \
    --run_name Qwen2-VL-2B-GRPO-CLEVR-70k \
    --save_steps 100 \
    --save_only_model true \
    --num_generations 8

Package                  Version     Editable project location
------------------------ ----------- ----------------------------------------------------------------
absl-py                  2.1.0
accelerate               1.3.0
aenum                    3.1.15
aiohappyeyeballs         2.4.4
aiohttp                  3.11.12
aiosignal                1.3.2
annotated-types          0.7.0
antlr4-python3-runtime   4.13.2
attrs                    25.1.0
av                       14.1.0
bitsandbytes             0.45.2
black                    25.1.0
blis                     0.7.11
catalogue                2.0.10
certifi                  2025.1.31
chardet                  5.2.0
charset-normalizer       3.4.1
click                    8.1.8
cloudpathlib             0.16.0
colorama                 0.4.6
colorlog                 6.9.0
confection               0.1.5
cymem                    2.0.11
DataProperty             1.1.0
datasets                 3.2.0
deepspeed                0.15.4
dill                     0.3.8
docker-pycreds           0.4.0
einops                   0.8.0
filelock                 3.17.0
flake8                   7.1.1
flash-attn               2.6.0
frozenlist               1.5.0
fsspec                   2024.9.0
gitdb                    4.0.12
GitPython                3.1.44
hf_transfer              0.1.9
hjson                    3.1.0
huggingface-hub          0.28.1
idna                     3.10
iniconfig                2.0.0
inquirerpy               0.3.4
isort                    6.0.0
Jinja2                   3.1.5
joblib                   1.4.2
langcodes                3.5.0
language_data            1.3.0
latex2sympy2_extended    1.0.6
liger_kernel             0.5.2
lighteval                0.7.0
lxml                     5.3.0
marisa-trie              1.2.1
markdown-it-py           3.0.0
MarkupSafe               3.0.2
math-verify              0.5.2
mbstrdecoder             1.1.4
mccabe                   0.7.0
mdurl                    0.1.2
mpmath                   1.3.0
msgpack                  1.1.0
multidict                6.1.0
multiprocess             0.70.16
murmurhash               1.0.12
mypy-extensions          1.0.0
networkx                 3.4.2
ninja                    1.11.1.3
nltk                     3.9.1
numpy                    1.26.4
nvidia-cublas-cu12       12.1.3.1
nvidia-cuda-cupti-cu12   12.1.105
nvidia-cuda-nvrtc-cu12   12.1.105
nvidia-cuda-runtime-cu12 12.1.105
nvidia-cudnn-cu12        9.1.0.70
nvidia-cufft-cu12        11.0.2.54
nvidia-curand-cu12       10.3.2.106
nvidia-cusolver-cu12     11.4.5.107
nvidia-cusparse-cu12     12.1.0.106
nvidia-cusparselt-cu12   0.6.2
nvidia-nccl-cu12         2.20.5
nvidia-nvjitlink-cu12    12.4.127
nvidia-nvtx-cu12         12.1.105
open-r1                  0.1.0.dev0  /nfs/code/R1-V/src/open-r1-multimodal/src
packaging                24.2
pandas                   2.2.3
parameterized            0.9.0
pathspec                 0.12.1
pathvalidate             3.2.3
pfzy                     0.3.4
pillow                   11.1.0
pip                      25.0
platformdirs             4.3.6
pluggy                   1.5.0
portalocker              3.1.1
preshed                  3.0.9
prompt_toolkit           3.0.50
propcache                0.2.1
protobuf                 3.20.3
psutil                   6.1.1
py-cpuinfo               9.0.0
pyarrow                  19.0.0
pycodestyle              2.12.1
pycountry                24.6.1
pydantic                 2.10.6
pydantic_core            2.27.2
pyflakes                 3.2.0
Pygments                 2.19.1
pytablewriter            1.2.1
pytest                   8.3.4
python-dateutil          2.9.0.post0
pytz                     2025.1
PyYAML                   6.0.2
qwen-vl-utils            0.0.10
regex                    2024.11.6
requests                 2.32.3
rich                     13.9.4
rouge_score              0.1.2
sacrebleu                2.5.1
safetensors              0.5.2
scikit-learn             1.6.1
scipy                    1.15.1
sentencepiece            0.2.0
sentry-sdk               2.20.0
setproctitle             1.3.4
setuptools               75.8.0
six                      1.17.0
smart-open               6.4.0
smmap                    5.0.2
spacy                    3.7.2
spacy-legacy             3.0.12
spacy-loggers            1.0.5
srsly                    2.5.1
sympy                    1.13.1
tabledata                1.3.4
tabulate                 0.9.0
tcolorpy                 0.1.7
tensorboardX             2.6.2.2
termcolor                2.3.0
thinc                    8.2.5
threadpoolctl            3.5.0
tokenizers               0.21.0
torch                    2.4.1
torchvision              0.19.1
tqdm                     4.67.1
transformers             4.49.0.dev0
triton                   3.0.0
trl                      0.15.0.dev0
typepy                   1.3.4
typer                    0.9.4
typing_extensions        4.12.2
tzdata                   2025.1
urllib3                  2.3.0
wandb                    0.18.3
wasabi                   1.1.3
wcwidth                  0.2.13
weasel                   0.3.4
wheel                    0.45.1
xxhash                   3.5.0
yarl                     1.18.3

**Comment by zxduan90** [comment]
不需要迭代那么多步，对于这种数据集，基本200个step，基本就够了。

**Comment by TobiasLee** [comment]
The results we reported are models trained after 100 steps (~ 30 minutes), which is close to your 36-minute training time.

**Comment by flashcp** [comment]
> 不需要迭代那么多步，对于这种数据集，基本200个step，基本就够了。

您是指这个任务比较简单吗？如果只跑200step，按batchsize=1，也才200张图，这样也能拟合这个数据吗

**Comment by TIMMY-CHAN** [comment]
我用4张A800，完整的17500个step需要251个小时
